src/dbnode/encoding/multi_reader_iterator.go

```diff
@@ -246,3 +246,7 @@ func (it *singleSlicesOfSlicesIterator) Size() (int, error) {
 	}
 	return size, nil
 }
+
+func (it *singleSlicesOfSlicesIterator) Rewind() {
+	it.firstNext = true
```

Comment:
This also needs to set `it.closed = false`; can we similarly add a test here?

Reply:
I think we don't actually want to un-Close something because that actually would result in it not actually being re-iterable.